test/unit/api/interactivity.test.js

```diff
@@ -1,15 +1,72 @@
+import Dataset from '../../../src/api/source/dataset';
+import Style from '../../../src/api/style';
+import Layer from '../../../src/api/layer';
 import Interactivity from '../../../src/api/interactivity';
 
 describe('api/interactivity', () => {
     describe('constructor', () => {
+        let source, style, layer, mockIntegrator;
+
+        beforeEach(() => {
+            source = new Dataset('ne_10m_populated_places_simple', {
+                user: 'test',
+                apiKey: '1234567890'
+            });
+            style = new Style();
+            layer = new Layer('layer', source, style);
+            mockIntegrator = { on: () => {} };
```

Comment:
you could use `jasmine.createSpy('on').and.returnValue({})`

Reply:
I don't need a spy